django/core/validators.py

```diff
@@ -128,7 +134,7 @@ def __call__(self, value):
                 raise
         else:
             # Now verify IPv6 in the netloc part
-            host_match = re.search(r'^\[(.+)\](?::\d{2,5})?$', urlsplit(value).netloc)
+            host_match = re.search(r'^\[(.+)\](?::\d{2,5})?$', self.split_url(value).netloc)
```

Comment:
Reverting this change and the one below doesn't result in any test failures. We should have tests for these cases too if they are indeed necessary.